cmd/thanos/rule.go

```diff
@@ -472,49 +490,22 @@ func runRule(
 	}
 
 	// Handle reload and termination interrupts.
-	reload := make(chan struct{}, 1)
 	{
 		cancel := make(chan struct{})
-		reload <- struct{}{} // Initial reload.
-
 		g.Add(func() error {
+			// Initialize rules.
+			if err := reloadRules(logger, ruleFiles, ruleMgr, evalInterval, metrics); err != nil {
+				level.Error(logger).Log("msg", "initialize rules failed", "err", err)
```

Comment:
For the initial load, we can actually fail and return the combined multierror.

Reply:
I don't have strong opinions about it, both works for me. In any case, I've pinged @bwplotka on the thread for the final decision.

Reply:
Yes, initial can fail indeed (:

Reply:
@bwplotka @kakkoyun Let's focus on the reload process first in this PR. Since the failure in iniliazation is breaking change.
After the reload related code is merged, I will start a new PR for the initialization failure only.
Any suggestion?

Reply:
cool, happy with that.